mteb/abstasks/AbsTaskInstructionRetrieval.py

```diff
@@ -580,56 +602,59 @@ def create_qrel_diff(self, og_qrels, changed_qrels):
-    total_length, num_pairs = calculate_length_and_count(
-        relevant_docs, queries, corpus, instructions
-    )
-    average_length = total_length / num_pairs if num_pairs else 0
-    num_documents = len(queries) + len(corpus)
-
-    language_description = f" for language {lang}" if lang else ""
-    print(
-        f"Average character length for changed{language_description} is {average_length}"
-    )
-    print(
-        f"Number of queries and documents{language_description} is {num_documents} (repeated 2x)"
-    )
-
-
-def calculate_length_and_count(relevant_docs, queries, corpus, instructions):
-    total_length = 0
-    num_pairs = 0
-    for query_id, docs in relevant_docs.items():
-        query = queries[query_id]
-        query += " " + instructions[query]
-        for doc_id in docs:
-            # not relevant
-            if docs[doc_id] == 0:
-                continue
-            doc = corpus[doc_id]
-            doc_text = doc["title"] + doc["text"]
-            total_length += len(query) + len(doc_text)
-            num_pairs += 1
-    return total_length, num_pairs
+    def _calculate_metrics_from_split(
+        self, split: str, lang: str | None = None
+    ) -> InstructionRetrievalDescriptiveStatistics:
+        if lang:
+            corpus = self.corpus[lang][split]
+            queries = self.queries[lang][split]
+            relevant_docs = self.og_relevant_docs[lang][split]
+            og_instructions = self.og_instructions[lang][split]
+            changed_instructions = self.changed_instructions[lang][split]
+            top_ranked = self.top_ranked[lang][split]
+        else:
+            corpus = self.corpus[split]
+            queries = self.queries[split]
+            relevant_docs = self.og_relevant_docs[split]
+            og_instructions = self.og_instructions[split]
+            changed_instructions = self.changed_instructions[split]
+            top_ranked = self.top_ranked[split]
+
+        total_corpus_len = sum(
+            [len(doc.get("title", "")) + len(doc["text"]) for doc in corpus.values()]
+        )
+        total_queries_len = sum([len(query) for query in queries.values()])
+        total_instructions_len = sum(
+            [len(instruction) for instruction in og_instructions.values()]
+        )
+        total_changed_instructions_len = sum(
+            [len(instruction) for instruction in changed_instructions.values()]
+        )
+        num_qrels_non_zero = sum(
+            sum(1 for doc_id in docs if docs[doc_id] != 0)
+            for docs in relevant_docs.values()
+        )
+        qrels_per_doc = num_qrels_non_zero / len(relevant_docs) if len(queries) else 0
+        top_ranked_per_query = (
+            sum(len(docs) for docs in top_ranked.values()) / len(queries)
+            if len(queries)
+            else 0
+        )
+        return {
+            "num_docs": len(corpus),
+            "num_samples": len(queries),
```

Comment:
Would probably rather have num_queries (which means I would move num_samples out of general dict)